tdrs-backend/tdpservice/search_indexes/parsers/tanf_parser.py

```diff
@@ -0,0 +1,269 @@
+            ('transportation_amount', 4, 72, 75, "Numeric"),
+            ('transport_nbr_months', 3, 76, 78, "Numeric"),
+            ('transition_services_amount', 4, 79, 82, "Numeric"),
+            ('transition_nbr_months', 3, 83, 85, "Numeric"),
+            ('other_amount', 4, 86, 89, "Numeric"),
+            ('other_nbr_months', 3, 90, 92, "Numeric"),
+            ('reduction_amount', 4, 93, 96, "Numeric"),
+            ('reduc_work_requirements', 1, 97, 97, "Numeric"),
+            ('reduc_adult_no_hs_diploma', 1, 98, 98, "Numeric"),
+            ('reduc_teen_not_in_school', 1, 99, 99, "Numeric"),
+            ('reduc_noncooperation_child_support', 1, 100, 100, "Numeric"),
+            ('reduc_irp_failure', 1, 101, 101, "Numeric"),
+            ('reduc_other_sanction', 1, 102, 102, "Numeric"),
+            ('reduc_prior_overpayment', 4, 103, 106, "Numeric"),
+            ('total_reduc_amount', 4, 107, 110, "Numeric"),
+            ('reduc_family_cap', 1, 111, 111, "Numeric"),
+            ('reduc_length_of_assist', 1, 112, 112, "Numeric"),
+            ('other_non_sanction', 1, 113, 113, "Numeric"),
+            ('waiver_control_grps', 1, 114, 114, "Numeric"),
+            ('tanf_family_exempt_time_limits', 2, 115, 116, "Numeric"),
+            ('tanf_new_child_only_family', 1, 117, 117, "Numeric"),
+            ('blank', 39, 118, 156, "Spaces"),
+        ]
+    )
+
+    # create search_index model
+    t1 = T1()
+    content_is_valid = True
+    for field in family_case_schema.get_all_fields():
+        if field.name == 'blank':
+            break
+        content = line[field.start-1:field.end]  # descriptor pdfs were off by one, could also adjust start values
+        if len(content) != field.length:
+            logger.warn('[LineNo:%d, col%d] Expected field "%s" with length %d, got: "%s"',
+                        line_number, field.start-1, field.name, field.length, content)
+            content_is_valid = False
+            continue
+        # check if content is type string or integer
+        if field.type == 'Numeric':
+            try:
+                content = int(content)
+            except ValueError:
+                logger.warn('[LineNo:%d, col%d] Expected field "%s" to be numeric, got: "%s"',
+                            line_number, field.start-1, field.name, content)
+                content_is_valid = False
+                continue
+        elif field.type == 'Alphanumeric':
+            pass  # maybe we can regex check some of these later
+        # The below is extremely spammy, turn on selectively.
+        # logger.debug('field: %s\t::content: "%s"\t::end: %s', field.name, content, field.end)
+
+        if content_is_valid:
+            setattr(t1, field.name, content)
+
+    if content_is_valid is False:
+        logger.warn('Content is not valid, skipping model creation.')
+        return
+
+        '''
+        '''
+
+    # try:
+    # t1.full_clean()
+    t1.save()
```

Comment:
Do we have to save t1 even for 'blank'?

Reply:
The 'blank' field is the last of the file so it breaks the loop and we then want to save everything we got to DB.